eras/alonzo/impl/src/Cardano/Ledger/Alonzo/Tools.hs

```diff
@@ -175,15 +169,16 @@ evaluateTransactionExecutionUnits pp tx utxo ei sysS costModels = do
     txb = getField @"body" tx
     ws = getField @"wits" tx
     dats = unTxDats $ getField @"txdats" ws
-    scripts = txscripts utxo tx
+    scriptsAvailable = txscripts utxo tx
     needed = scriptsNeeded utxo tx
+    nonPhase1Needed = mapMaybe (knownToNotBe1Phase scriptsAvailable) needed
```

Comment:
Same thing here. `onlyPlutusScripts`

Reply:
I was trying to stress that we might not have _all_ the plutus scripts. we've just gathered the ones we can, based on what we know. `onlyPlutusScripts` is mathematically correct, but maybe elides the presence of sneaky scripts we can't identify?

Reply:
yep, `txscripts` gathers all the scripts that we have available for executing (both simple and Plutus). Some of which we don't want to execute, because they are not actually needed. The ones we need are identified by a collection of hashes, namely `needed`. In order to figure out which of the needed one are Plutus scripts, we look them up in `scriptsAvailable`.  The sneaky part is that there could be a script hash that corresponds to a script that we need, but that we cannot identify as being a Plutus script since it is not in `scriptsAvailable`.

Reply:
I've gone with `neededAndConfirmedToBePlutus` for this variable and the one above that we talked about.